keylime/enrich.py

```diff
@@ -0,0 +1,289 @@
+import re
+import traceback
+from ctypes import CDLL, create_string_buffer, byref, c_char_p
+
+import yaml
+try:
+    from yaml import CSafeLoader as SafeLoader
+except ImportError:
+    from yaml import SafeLoader
+
+##################################################################################
+#
+# yaml by default outputs numbers in decimal format, and this allows us to
+# represent some numbers in hexadecimal
+#
+##################################################################################
+
+
+class hexint(int):
+    pass
+
+
+def representer(_, data):
+    return yaml.ScalarNode('tag:yaml.org,2002:int', hex(data))
+
+
+yaml.add_representer(hexint, representer)
+
+##################################################################################
+#
+# Binary data from TSS will be in Little Endian according to the TPM spec, so
+# we need to convert them to host byte order before hand
+#
+##################################################################################
+
+##################################################################################
+#
+# These function use efivar C libraries to decode device path and guid
+#
+##################################################################################
+
+efivarlib = "libefivar.so"  # formerly "/usr/lib/x86_64-linux-gnu/libefivar.so"
+
+efivarlib_functions = CDLL(efivarlib)
+
+
+def getDevicePath(b):
+    ret = efivarlib_functions.efidp_format_device_path(0, 0, b, len(b))
+    if ret < 0:
+        raise Exception(
+            f'getDevicePath: efidp_format_device_path({b}) returned {ret}')
+
+    s = create_string_buffer((ret+1))
+
+    ret = efivarlib_functions.efidp_format_device_path(s, ret+1, b, len(b))
+    if ret < 0:
+        raise Exception(
+            f'getDevicePath: efidp_format_device_path({b}) returned {ret}')
+
+    return s.value.decode("utf-8")
+
+
+def getGUID(b):
+    s = c_char_p(None)
+    ret = efivarlib_functions.efi_guid_to_str(b, byref(s))
+    if ret < 0:
+        raise Exception(f'getGUID: efi_guid_to_str({b}) returned {ret}')
+    return s.value.decode("utf-8")
+
+##################################################################################
+#
+# https://uefi.org/sites/default/files/resources/UEFI%20Spec%202.8B%20May%202020.pdf
+# Section 32.4.1
+#
+# Relevant data structures
+#
+#  typedef struct _EFI_SIGNATURE_LIST {
+#       uint8_t SignatureType[16];
+#       uint32_t SignatureListSize;
+#       uint32_t SignatureHeaderSize;
+#       uint32_t SignatureSize;
+#       // uint8_t SignatureHeader[SignatureHeaderSize];
+#       // uint8_t Signatures[][SignatureSize];
+#  } EFI_SIGNATURE_LIST;
+#
+#  typedef struct _EFI_SIGNATURE_DATA {
+#      uint8_t SignatureOwner[16];
+#      uint8_t SignatureData[];
+#  } EFI_SIGNATURE_DATA;
+#
+##################################################################################
+
+##################################################################################
+# Print out EFI_SIGNATURE_DATA
```

Comment:
This method doesn't print anything, it returns a structure of the owner and data fields

Reply:
fixed